provider.go

```diff
@@ -0,0 +1,95 @@
+package joohoi_acme_dns
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/libdns/libdns"
+)
+
+var acmePrefix = "_acme-challenge."
+
+type DomainConfig struct {
+	Username   string `json:"username,omitempty"`
+	Password   string `json:"password,omitempty"`
+	Subdomain  string `json:"subdomain,omitempty"`
+	FullDomain string `json:"fulldomain,omitempty"`
+	ServerURL  string `json:"server_url,omitempty"`
+}
+
+type Domain = string
```

Comment:
```suggestion
type Domain string
```

(Could probably just use a `string` too, since there's no behavior associated with this type.)

Reply:
I just wanted to make clear that the mapping is from domain to `DomainConfig` But I guess that's clear enough from the docs and context, I will just go with `string`.

Reply:
Changed: https://github.com/libdns/libdns/pull/64/commits/fbf6d6cdc86fd94ceec785b4dcb741319d4511c0